packages/snaps-sdk/src/internals/structs.ts

```diff
@@ -97,6 +98,7 @@ export function typedUnion<Head extends AnyStruct, Tail extends AnyStruct[]>(
         : struct,
     )
     .flat(Infinity);
+  const types = flatStructs.map(({ schema }) => schema.type.type);
```

Comment:
Why did you move this?

Reply:
Under the assumption that it would be cheaper computationally to only compute it once per struct.